pricegraph/src/orderbook.rs

```diff
@@ -359,20 +359,21 @@ impl Orderbook {
     ///
     /// # Panics
     ///
-    /// This method panics if the provided flow is empty and would not further
-    /// reduce the orderbook. This can happen in certain situations where the
-    /// flow becomes empty due to floating point rounding errors and can cause
-    /// infinite loops when reducing orderbook or enumerating transitive orders.
-    fn fill_path_with_flow(&mut self, path: &[NodeIndex], flow: &Flow) -> Option<()> {
-        if flow.is_empty() {
-            panic!("stuck reducing path with empty flow: {}", format_path(path));
-        }
-
+    /// If an order along the path doesn't exist.
+    fn fill_path_with_flow(
+        &mut self,
+        path: &[NodeIndex],
+        flow: &Flow,
+    ) -> Result<(), OrderbookError> {
         let mut transitive_xrate = ExchangeRate::IDENTITY;
         for pair in pairs_on_path(path) {
-            let (order, user) = self.best_order_with_user_for_pair_mut(pair)?;
+            let (order, user) = self
+                .best_order_with_user_for_pair_mut(pair)
+                .unwrap_or_else(|| panic!("missing order for pair {:?}", pair));
```

Comment:
This function used to return Option<()> where the None case was treated as the error and this error would always have been unwrapped by the caller. The change in behavior is only that I moved the unwrap into here. This is cleaner because this function now returns a Result that we don't want to unwrap.